mlir/test/mlir-miopen-driver/padding_map.mlir

```diff
@@ -0,0 +1,17 @@
+// This tests checks the  affinemap component:
+// RUN: mlir-opt -miopen-lowering -miopen-affine-transform %s | FileCheck %s
+
+module  {
+  func @pad_parameters_00(%arg0: memref<256x1x128x28x28xf32>) attributes {kernel} {
+   %1 = miopen.transform(%arg0) {extraPad = "false", gemmK_extra = 0 : i32, gemmN_extra = 0 : i32, layout = [{dimensions = [1 : i32], names = ["gi"], source_dimensions = [1 : i32], source_names = ["gi"], transformation = "PassThrough"}, {dimensions = [0 : i32], names = ["ni"], source_dimensions = [0 : i32], source_names = ["ni"], transformation = "PassThrough"}, {dimensions = [2 : i32], names = ["ci"], source_dimensions = [2 : i32], source_names = ["ci"], transformation = "PassThrough"}, {dimensions = [3 : i32, 4 : i32], names = ["hipad", "wipad"], parameters = [1 : i32, 1 : i32], source_dimensions = [3 : i32, 4 : i32], source_names = ["hi", "wi"], transformation = "Pad"}], output_layout = ["ni", "gi", "ci", "hipad", "wipad"], source_layout = ["ni", "gi", "ci", "hi", "wi"]} : memref<256x1x128x28x28xf32> to memref<256x1x128x28x28xf32>
+  return
+  }
+
+  func @pad_parameters_1101(%arg0: memref<256x1x128x28x28xf32>) attributes {kernel} {
+   %1 = miopen.transform(%arg0) {extraPad = "false", gemmK_extra = 0 : i32, gemmN_extra = 0 : i32, layout = [{dimensions = [1 : i32], names = ["gi"], source_dimensions = [1 : i32], source_names = ["gi"], transformation = "PassThrough"}, {dimensions = [0 : i32], names = ["ni"], source_dimensions = [0 : i32], source_names = ["ni"], transformation = "PassThrough"}, {dimensions = [2 : i32], names = ["ci"], source_dimensions = [2 : i32], source_names = ["ci"], transformation = "PassThrough"}, {dimensions = [3 : i32, 4 : i32], names = ["hipad", "wipad"], parameters = [1 : i32, 1 : i32, 0 : i32, 1 : i32], source_dimensions = [3 : i32, 4 : i32], source_names = ["hi", "wi"], transformation = "Pad"}], output_layout = ["ni", "gi", "ci", "hipad", "wipad"], source_layout = ["ni", "gi", "ci", "hi", "wi"]} : memref<256x1x128x28x28xf32> to memref<256x1x128x28x28xf32>
+  return
+  }
+}
+
+// CHECK: affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2, d3 - 1, d4 - 1)>
```

Comment:
since the math is pretty involved, please move `CHECK` macro next to corresponding `miopen.transform` op, and add comments to explain why.

Reply:
also, there is no case to check `extraPad = "true"`.